src/schema.ts

```diff
@@ -182,17 +199,80 @@ export class Schema {
 }
 
 /**
- * @return a summary with the correct distinct property given a max number of bins
+ * @return a summary of the binning scheme determined from the given max number of bins
  */
-function binSummary(maxbins: number, summary: Summary) {
-  const bin = dlBin({
+function binSummary(maxbins: number, summary: Summary): Summary {
+  const bin: Bin = dl.bins({
     min: summary.min,
     max: summary.max,
     maxbins: maxbins
   });
-  const binSum = extend({}, summary);
-  binSum.distinct = (bin.stop - bin.start) / bin.step;
-  return binSum;
+
+  return binStats(bin, summary);
+}
+
+function timeUnitSummary(unit: TimeUnit, summary: Summary, type: Type): Summary {
+  var bin: Bin;
+  switch (unit) {
+    // these are the units that dl.bins.date supports
```

Comment:
To convert YEARMONTH to months and same for others, could you help add `toDataLibTimeUnit` in VL's `timeunit.ts`?   This would might become helpful for other projects.  

- [ ] Also notice that `hours` = `hour` in DL,  `datehours` = `hours`, `day` = `weekdays`, et.c